.github/workflows/create-release.yml

```diff
@@ -49,13 +42,18 @@ jobs:
           file: "v2/out/release/*"
           file_glob: true
 
-      - name: Login to registry
-        # note that all creds are on host and never passed into devcontainer
-        uses: docker/login-action@v2.1.0
-        with:
-          registry: ${{ secrets.REGISTRY_LOGIN }}
-          username: ${{ secrets.AZURE_CLIENT_ID }}
-          password: ${{ secrets.AZURE_CLIENT_SECRET }}
+      - name: Docker login
+        run: |
+          container_id=${{env.container_id}}
+          docker exec -e AZURE_CLIENT_ID -e AZURE_CLIENT_SECRET -e DOCKER_REGISTRY "$container_id" task docker-login
```

Comment:
Now that these secrets are being passed in to the devcontainer, is there any possibility of them being logged or left behind as a part of the published image? If so, this opens up the chance of someone harvesting those secrets.

Reply:
I don't think so - We already do pass-in secrets for live validation and az-login in other workflows as well.

Reply:
Checking a `git blame` of this file, it was @Porges who wrote the comment

> `# note that all creds are on host and never passed into devcontainer`

Working from the precept that smart people do things for good reasons, I'm worried that we're missing something here. To be more specific on my earlier concern, I'm not worried about disclosure during CI, I'm worried about the credentials being left lying around _inside the container image_, available for anyone to extract if they go nosing around inside.

Reply:
> By default, Docker looks for the native binary on each of the platforms, i.e. "osxkeychain" on macOS, "wincred" on windows, and "pass" on Linux. A special case is that on Linux, Docker will fall back to the "secretservice" binary if it cannot find the "pass" binary. If none of these binaries are present, it stores the credentials (i.e. password) in base64 encoding in the config files described above.

Here's the [reference](https://docs.docker.com/engine/reference/commandline/login/#default-behavior) to how docker stores the credentials and auth tokens.